Formula/tdlib.rb

```diff
@@ -0,0 +1,50 @@
+class Tdlib < Formula
+  desc "Cross-platform library for building Telegram clients"
+  homepage "https://core.telegram.org/tdlib"
+  url "https://github.com/tdlib/td/archive/v1.3.0.tar.gz"
+  sha256 "2953fe75027ac531248b359123aa4812666377ac874c1db506fa74176f2d2338"
+  head "https://github.com/tdlib/td.git"
+
+  depends_on "cmake" => :build
+  depends_on "gperf"
+  depends_on "openssl"
+  depends_on "readline"
+
+  def install
+    mkdir "build" do
+      system "cmake", "-DCMAKE_BUILD_TYPE=Release", "-DOPENSSL_ROOT_DIR=/usr/local/Cellar/openssl/1.0.2p", ".."
```

Comment:
- `CMAKE_BUILD_TYPE` is part of `std_cmake_args`
- Why are your running `cmake` twice?
- Can cmake find openssl without the special flag?

Reply:
Re: cmake twice

Following the instructions in the parent project: https://github.com/tdlib/td/blob/master/README.md#building

Note the change from .. to ., it's almost like the project is set up to generate a cmake project in the build directory and then from there generate a makefile.

I was seeing link errors without the OPENSSL_ROOT_DIR define, see the commentary at https://github.com/tdlib/td/blob/master/README.md#macos

However, this feels really gross to be hardcoding the version of openssl and the path to my Cellar directory, are there some variables I should be using instead?

Reply:
`#{Formula["openssl"].opt_prefix}`

Reply:
:+1:

Reply:
Actually, looks like this wasn't necessary after all, my bad.